tests/test_device.py

```diff
@@ -340,6 +306,33 @@ def test_check_validity_on_valid_queue(self, mock_device_supporting_paulis):
         # Raises an error if queue or observables are invalid
         dev.check_validity(queue, observables)
 
+    def test_check_validity_containing_prod(self, mock_device_supporting_prod):
+        """Tests that the function Device.check_validity works with Prod"""
+
+        dev = mock_device_supporting_prod()
+
+        queue = [
+            qml.PauliX(wires=0),
+            qml.PauliZ(wires=1),
+        ]
+
+        observables = [
+            qml.expval(qml.PauliX(0) @ qml.PauliZ(1)),
+            qml.expval(qml.PauliZ(0) @ (qml.PauliX(1) @ qml.PauliZ(2))),
+        ]
+
+        dev.check_validity(queue, observables)
+
+        unsupported_nested_observables = [
+            qml.expval(qml.PauliZ(0) @ (qml.PauliX(1) @ qml.PauliY(2)))
```

Comment:
Perhaps I am standing on the hose but what is different here? 🤯

Reply:
The mock device I made did not support `PauliY`, so this is to test that a `Prod` of `PauliZ` and another `Prod` is handled properly, that is, the whole thing unwraps, and support for each operand is checked.

Reply:
Aaaaah I missed the final op being a *Y*, thanks!